module_utils/network_lsr/argument_validator.py

```diff
@@ -2216,6 +2217,15 @@ def _ipv6_is_not_configured(connection):
                     idx,
                     "ip.ipv6_disabled is not supported by initscripts.",
                 )
+
+            if not self.TOX_UNIT_TEST and not hasattr(
```

Comment:
What's the problem that `self.TOX_UNIT_TEST` is supposed to fix? You need to find another solution, maybe adjusting the unit test to expect something else or to mock or remove the `"SETTING_IP6_CONFIG_METHOD_DISABLED"`

Reply:
`Util.NM()` called the function here, 
https://github.com/linux-system-roles/network/blob/main/module_utils/network_lsr/utils.py#L65. 
Under the tox Python2 test virtual env, we can not import the `gi` module or install  pygobject in proper manner. So I propose to test differently regarding to the test env:

```
    def test_ipv6_dns_with_ipv6_disabled(self):
        """
        Test that configuring IPv6 DNS is not allowed when IPv6 is disabled.
        """
        validator = network_lsr.argument_validator.ArgValidator_ListConnections()
        ipv6_dns_with_ipv6_disabled = [
            {
                "name": "test_ipv6_dns",
                "type": "ethernet",
                "ip": {
                    "ipv6_disabled": True,
                    "dhcp4": True,
                    "dns": ["2001:db8::20"],
                },
            }
        ]

        if sys.version[:3] >= "3.6" and sys.version[3] == ".":
            self.assertRaisesRegex(
                ValidationError,
                "IPv6 needs to be enabled to support IPv6 nameservers.",
                validator.validate_connection_one,
                "nm",
                validator.validate(ipv6_dns_with_ipv6_disabled),
                0,
            )
        else:
            self.assertEqual(
                validator.validate(ipv6_dns_with_ipv6_disabled)[0]["ip"][
                    "ipv6_disabled"
                ],
                True,
            )
```